app/models/booking.rb

```diff
@@ -2,4 +2,14 @@ class Booking < ApplicationRecord
   belongs_to :user
   belongs_to :article
   has_one :review
+  validates :start_date, presence: true
+  validates :end_date, presence: true
+  validate :start_must_be_before_end_date
+
```

Comment:
This is done on the article view page. The booking form is only displayed if the article is not currently accessible